hype-caplet/src/main/java/Hypelet.java

```diff
@@ -179,6 +188,11 @@ private static String getRunId() {
   }
 
   private static FileSystemProvider loadFileSystemProvider(URI uri) throws IOException {
-    return FileSystems.newFileSystem(uri, emptyMap(), Hypelet.class.getClassLoader()).provider();
+    if (Objects.equals(uri.getScheme(), "file")) {
+      return FileSystems.getFileSystem(URI.create("file:///")).provider();
+    } else {
+      // TODO: check if I can just getFileSystem for all schemes?
```

Comment:
The reason this had to go through `FileSystems.newFileSystem` was because the gsc-nio jar is only loaded as part of the Hypelet jar which is downloaded with the Maven Caplet. So it's not part of the classpath initially.